docs/developer/extending/apps/developing-apps/troubleshooting.mdx

```diff
@@ -2,15 +2,30 @@
 title: Troubleshooting
 ---
 
-Find solutions to common issues that you may face during apps development.
+Find solutions to common issues that you may face during app development.
+
 # Issues
+
 ## I see Typescript errors
 
 Saleor [Apps](https://github.com/saleor/apps), [SDK](https://github.com/saleor/saleor-app-sdk) and [App Template](https://github.com/saleor/saleor-app-template/discussions)
-use [pnpm](https://pnpm.io/) package manager. It can be sensitive with dependencies resolutions, especially in a monorepo setup.
+use [pnpm](https://pnpm.io/) package manager. It can be sensitive to dependencies resolutions, especially in a monorepo setup.
 
 Please ensure you have installed the right version of pnpm on your machine and re-install dependencies with `pnpm install`.
 
 To find the accurate pnpm version in your project, check the root `package.json` file and look for the `packageManager` field.
 
 For example: `"packageManager": "pnpm@8.2.0"` means that you need to install exactly the `8.2.0` version of pnpm.
+
+## Client-side GraphQL requests are rejected due to CORS error
+
+If you are using the Saleor Cloud instance, you may encounter the following error when trying to make a GraphQL request from the client side:
+
+```
+Access to fetch at 'https://xxx.eu.saleor.cloud/' from origin 'https://your-application' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
+```
+
+The issue might be caused by the CORS policy set up in the Saleor Cloud dashboard. Navigate to the environment details of your instance and check the `Allowed API origins` section. If option `Selected origins` is selected, you have two possible solutions:
```

Comment:
> Navigate to the environment details of your instance and check the `Allowed API origins` section.

**suggestion**: I think we could use a bit more descriptive instruction here. Can you tell where exactly is this "Allowed API origins" section?

Reply:
@peelar I think its pretty obvious when you go to cloud console